src/storage/storage.ts

```diff
@@ -1,46 +1,44 @@
-import { ActivityPub } from "../activityPub/activityPub";
-import { BatchFileObject } from "../batch/batch";
-// import * as config from "../config/config";
-import { Config } from "../config/config";
-import { NotImplementedError } from "../utilities/errors";
+import { Config, getConfig } from "../config/config";
+import { MissingStore } from "../utilities/errors";
 
-export type File = BatchFileObject | ActivityPub;
+export type File = Buffer | string;
+export type Content = string | Buffer;
 
 export interface StorageInterface {
-  store(file: File): Promise<string>;
-  fetch(uri: string): Promise<File>;
+  put(targetPath: string, content: Content): Promise<URL>;
+  get(targetPath: string): Promise<File>;
 }
 
 /**
- * store() takes a batch file to store with the chosen hosting solution and
- * returns the URI of the file once stored. This method is not yet implemented.
+ * put() takes a batch file to store with the chosen hosting solution and
+ * returns the URI of the file once stored.
  *
- * @param   file  The file object to store on the chosen hosting solution
+ * @param   _targetPath  The location where the file is stored
+ * @param   _content The file object to store on the chosen hosting solution
  * @param   opts  Any optional overrides for the default configuration settings
  * @returns       The URI of the hosted file
  */
-export const store = async (_file: File, _opts?: Config): Promise<string> => {
-  throw NotImplementedError;
+export const put = async (_targetPath: string, _content: Content, _opts?: Config): Promise<URL> => {
+  const { store } = await getConfig();
+  if (!store) throw MissingStore;
 
-  // const config = getConfig(opts);
-  // return await config.store.store(file);
+  return await store.put(_targetPath, _content);
 };
 
 /**
- * fetch() takes a URI and fetches a batch file from the given address. This
+ * get() takes a URI and fetches a batch file from the given address. This
  * method is not yet implemented.
  *
- * @param   uri   The URI of the batch file to download
- * @param   opts  Any optional overrides for the default configuration settings
+ * @param   _targetPath  The URI of the batch file to download
  * @returns       The batch file fetched from the given URI
  *
  * @throws HTTPError
  * Indicates an error was thrown at the HTTP layer. This may indicate that the
  * host has censored the content for legal reasons if the error code is 451.
  */
-export const fetch = async (_uri: string, _opts?: Config): Promise<File> => {
-  throw NotImplementedError;
+export const get = async (_targetPath: string): Promise<File> => {
```

Comment:
Why the underscore?

Reply:
I was not sure why this was underscored myself. I just kept it as is. However, now that you point it out I will update it.